Konnect Public Dev Portal (#2574)
Kong/docs.konghq.com: app/konnect/dev-portal/administrators/public-portal.md

---
title: Public Developer Portal
no_version: true
---

Enable public access to your Dev Portal. When Public Portal is enabled, developers

> don't need to register for the portal access and any new application registrations aren't

Reviewer: > new application registrations aren't allowed

Is there any security risk with existing application registrations? Or even if not, maybe this is something the admin should know: enabling public access means that app reg is disabled, but any existing registered applications will continue to work.

Author: > > new application registrations aren't allowed
> 
> Is there any security risk with existing application registrations? Or even if not, maybe this is something the admin should know: enabling public access means that app reg is disabled, but any existing registered applications will continue to work.

I'm not sure if there are any security risks. @skibum would have to answer that question. Also it seems confusing to say app reg is disabled when the button can stay enabled!

Reviewer 2: There is not a security risk for previously approved applications.

Author: Thanks for your response @skibum.

> Enable public access to your Dev Portal. When Public Portal is enabled, developers
> don't need to register for the portal access and any new application registrations aren't
> allowed. The Register options are not available in the portal.

Reviewer: ```suggestion
Enable public access to your Dev Portal. With **Public Portal** enabled, anyone 
can access your Dev Portal without a login. Developers don't need to register 
for portal access and new application registrations aren't allowed. 

The Register options are not available in a public-facing portal.
```
Calling out that this disables portal login _in general_ - this was a common point of confusion from the bug bash right before launch, as an admin would try to access the portal and find themselves needing a specific dev portal account.

![Konnect Settings for a Public Portal](/assets/images/docs/konnect/konnect-portal-auto-approve.png)

> ## Enable a Public Dev Portal

Reviewer: This reads a bit awkwardly. The goal/action is to make your Dev Portal public, right? So maybe the title should be "Make a Dev Portal public" or something along those lines.

Same for the title below: "Disable public access to a Dev Portal"? As a user, I'd read the title below as disabling a public Dev Portal entirely, ie taking it down, which isn't the case.

Author: > This reads a bit awkwardly. The goal/action is to make your Dev Portal public, right? So maybe the title should be "Make a Dev Portal public" or something along those lines.
> 
> Same for the title below: Change "Disable a Public Dev Portal" to "Disable public access to a Dev Portal"? As a user, I'd read the title below as disabling a public Dev Portal entirely, ie taking it down, which isn't the case.

To be consistent with the remainder of the portal docs, we should stick with "Enable" rather than "Make".  So went with "Enable public access to a Dev Portal" and "Disable  Public Access to a Dev Portal" as you suggested. Thanks Lena!
